### DISCUSSION TITLE
no matching peer config found

### DISCUSSION THREAD

**Discussion by SB8767** [strongswan/strongswan#2288] [Q&A]
Hi,

I'm trying to set up a test VPN connection to compare it against another VPN but I am getting no peer config found.

Config 1:
conn vpn-test
    type=tunnel
    auto=start
    keyexchange=ikev2
    authby=secret
    left=188.166.169.90
    leftsubnet=10.0.5.1/29
    leftid=186.166.169.90
    leftfirewall=yes
    right=*Publicip*
    rightsubnet=192.168.10.1/29
    rightid=10.131.112.99
    rightfirewall=yes
    ike=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha-modp2048!
    esp=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha1-modp2048!
    dpdaction=restart
    ikelifetime=86400s
    lifetime=3600s
    dpddelay=20s
    dpdtimeout=60s
    

ipsec.secrets output:
#vpn-test
188.166.169.90  10.131.112.99 : PSK "*key*"

config2:

conn vpn-test
    type=tunnel
    auto=start
    keyexchange=ikev2
    authby=secret
    left=**
    leftsubnet=192.168.10.1/29
    leftid=10.131.112.99
    leftfirewall=yes
    right=188.166.169.90
    rightsubnet=10.0.5.1/29
    rightfirewall=yes
    ike=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha-modp2048!
    esp=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha1-modp2048!
    dpdaction=restart
    ikelifetime=86400s
    lifetime=3600s
    dpddelay=20s
    dpdtimeout=60s

ipsec.secrets output:

#vpn-test
 publicip 188.166.169.90 : PSK "*key*"

when I run the below command from config1 server:

```
grep -rn /var/log/syslog -e "charon"
```

I get this:
```
ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
3685:2024-06-13T16:15:20.092942+00:00 ubuntu-s-1vcpu-1gb-lon1-01 charon: 06[CFG] looking for peer configs matching 188.166.169.90[188.166.169.90]...publicip[10.131.112.99]
3686:2024-06-13T16:15:20.093158+00:00 ubuntu-s-1vcpu-1gb-lon1-01 charon: 06[CFG] no matching peer config found
3687:2024-06-13T16:15:20.093318+00:00 ubuntu-s-1vcpu-1gb-lon1-01 charon: 06[IKE] peer supports MOBIKE
3688:2024-06-13T16:15:20.093525+00:00 ubuntu-s-1vcpu-1gb-lon1-01 charon: 06[ENC] generating IKE_AUTH response 1 [ N(AUTH_FAILED) ]
3689:2024-06-13T16:15:20.093791+00:00 ubuntu-s-1vcpu-1gb-lon1-01 charon: 06[NET] sending packet: from 188.166.169.90[4500] to publicip[4500] (96 bytes)

```

I was using this website as a reference: https://ruan.dev/blog/2018/02/11/setup-a-site-to-site-ipsec-vpn-with-strongswan-and-preshared-key-authentication

**Comment by tobiasbrunner** [comment]
> leftid=186.166.169.90

Typo

**Reply by SB8767** [reply]
perfect - that solved the issue and the VPN is running. Now I'm back to square one and trying to find out why I cannot ping 192.168.10.2 from 188.166.169.90/ 10.0.5.2 server


root@ubuntu-s-1vcpu-1gb-lon1-01:/etc/strongswan.d# ping 192.168.10.2
PING 192.168.10.2 (192.168.10.2) 56(84) bytes of data.|
--- 192.168.10.2 ping statistics ---
13 packets transmitted, 0 received, 100% packet loss, time 12312m

this is the interface:
eth0:2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.10.2  netmask 255.255.255.248  broadcast 192.168.10.7
        ether 16:a7:4d:68:fa:2f  txqueuelen 1000  (Ethernet)

**Reply by SB8767** [reply]
See below outputs from traceroute:


188.166.169.90 server:


traceroute to 192.168.10.2 (192.168.10.2), 30 hops max, 60 byte packets
 1  * * *
 2  10.80.71.5 (10.80.71.5)  3.429 ms 10.80.70.253 (10.80.70.253)  3.145 ms 10.80.71.7 (10.80.71.7)  3.063 ms
 3  138.197.249.110 (138.197.249.110)  3.582 ms 138.197.249.126 (138.197.249.126)  3.547 ms 138.197.249.102 (138.197.249.102)  3.802 ms
 4  143.244.192.42 (143.244.192.42)  3.780 ms 143.244.192.46 (143.244.192.46)  3.264 ms 143.244.192.42 (143.244.192.42)  3.241 ms
 5  143.244.224.144 (143.244.224.144)  3.420 ms  2.988 ms 143.244.224.142 (143.244.224.142)  3.376 ms
 
 other server:
 
  traceroute 10.0.5.2
traceroute to 10.0.5.2 (10.0.5.2), 30 hops max, 60 byte packets
 1  * * *
 2  10.80.71.5 (10.80.71.5)  1.886 ms  1.713 ms  1.854 ms
 3  138.197.249.108 (138.197.249.108)  1.716 ms 138.197.249.100 (138.197.249.100)  2.177 ms 138.197.249.118 (138.197.249.118)  1.803 ms
 4  143.244.192.46 (143.244.192.46)  1.300 ms  1.392 ms 143.244.192.38 (143.244.192.38)  1.217 ms
 5  143.244.224.150 (143.244.224.150)  1.360 ms 143.244.224.144 (143.244.224.144)  1.550 ms  1.707 ms

**Reply by tobiasbrunner** [reply]
Could be so many things, firewall, routing, config. You have to provide a lot more information (#196), or try to debug this yourself by looking at the relevant status output etc.

**Reply by SB8767** [reply]
Sorry.

Both servers have the same firewall attached. UFW is disabled.

A side has the below POSTROUTING rule:
iptables -t nat -A POSTROUTING -s 10.0.5.1/29 -d 192.168.10.1/29 -j MASQUERADE

A side IP address is on eth0:2
eth0:2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.5.2  netmask 255.255.255.248  broadcast 10.0.5.7
        ether 3a:24:3f:c0:7b:c9  txqueuelen 1000  (Ethernet)


B side has the below POSTROUTING rule:
iptables -t nat -A POSTROUTING -s 192.168.10.1/29 -d 10.0.5.0/29 -j MASQUERADE

B side IP address is on eth0:2
eth0:2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.10.2  netmask 255.255.255.248  broadcast 192.168.10.7
        ether 16:a7:4d:68:fa:2f  txqueuelen 1000  (Ethernet)
        
However, from the A side I can the B side leftid (10.131.112.99).
    
 I can also ping A side LAN IP from B side 
    ping 10.131.51.48
PING 10.131.51.48 (10.131.51.48) 56(84) bytes of data.
64 bytes from 10.131.51.48: icmp_seq=1 ttl=64 time=2.65 ms
64 bytes from 10.131.51.48: icmp_seq=2 ttl=64 time=1.21 ms
64 bytes from 10.131.51.48: icmp_seq=3 ttl=64 time=0.506 ms
64 bytes from 10.131.51.48: icmp_seq=4 ttl=64 time=0.779 ms


This tells me both side can ping the LAN IP of the other side but it cannot ping the specific subnet that is set in ipsec.conf

**Reply by tobiasbrunner** [reply]
What's your intention with those MASQUERADE rules?

**Reply by SB8767** [reply]
to be honest, I thought it would do some nat'ing. I forgot to mention in my previous post I have ipv4_forward = 1 enabled on both servers. I still can't figure out why I can ping the LAN IP from either server, but not the eth0:2 IP addresses

**Reply by tobiasbrunner** [reply]
Well, I don't get what you are trying to achieve with those NAT rules that involve the subnets you configured as traffic selectors (`left|rightsubnet`) for the IPsec connection. The result will likely be that no packets (if they originally have addresses in these subnets) will match the IPsec policies after applying the NAT (the source address is probably replaced by some other address outside the source subnet). So again, what's your goal here?

**Reply by SB8767** [reply]
A side has a subnet of
`leftsubnet=10.0.5.1/29`

B side has a subnet of

`rightsubnet=192.168.10.1/29`

On either connection I cannot ping the other side i.e. I cannot ping B side subnet from side A, and vice versa. The VPN is up and stable.

Both A and B side have their subnet as virtual interfaces (eth0:1). 

Also, it's worth mentioning from B side I can ping the LAN IP of subnet of side A and vice versa. So pings work through the VPN tunnel but just not to the virtual interfaces.

**Reply by tobiasbrunner** [reply]
What are those LAN IPs? Those pings will most likely not go through the VPN as the traffic selectors won't match. And these are not actually "virtual interfaces", just additional IPs on the eth0 interface with a label that's interpreted by ifconfig (which is a deprecated tool, you should use iproute2 e.g. `ip addr` instead). So your goal is to be able to ping `192.168.10.1` from `10.0.5.1` and vice-versa? Without the NAT rules that change such packets and make them not match the IPsec policies, that should already be possible.

**Reply by SB8767** [reply]
The LAN IPs are just LAN IPs of each server. 

`So your goal is to be able to ping 192.168.10.1 from 10.0.5.1 and vice-versa?` yes. I deleted the NAT rules as they didn't solve the ping issue.

**Reply by tobiasbrunner** [reply]
Look at the packets and whether they match the IPsec policies. Look at traffic counters (strongSwan, firewall etc.).

**Reply by SB8767** [reply]
Thank you.

I did some further troubleshooting today:

1. I set up x2 servers and set up StrongSwan VPN on a different cloud provider -pings worked fine without the need for any NAT or IPTables.
2. I tried replicating this on my current Cloud provider but I could not ping.  I found the problem to be leftid and rightid - I removed both and restarted ipsec, and that worked. I could successfully ping.
3. My next step is: I will replicate this on the servers this case was logged for:

> A side has a subnet of
> leftsubnet=10.0.5.1/29
> 
> B side has a subnet of
> rightsubnet=192.168.10.1/29
> 

This is a working ipsec.conf on one of the servers with my current Cloud Provider:

```
conn vpn-test
    type=tunnel
    auto=start
    keyexchange=ikev2
    authby=secret
    left=143.110.169.236
    leftsubnet=192.168.100.2/24
    leftfirewall=yes
    right=165.232.100.250
    rightsubnet=192.168.200.2/24
    rightfirewall=yes
    ike=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha-modp2048!
    esp=aes256-sha512-modp2048,aes256-sha384-modp2048,aes256-sha256-modp2048,aes256-sha1-modp2048!
    dpdaction=restart
    ikelifetime=28800s
    lifetime=28800s
    dpddelay=20s
    dpdtimeout=60s
```

Admittedly I went the long way to troubleshoot but I believe the leftid and rightid may have been the problem and I should know m more in the next few hours.

**Reply by SB8767** [reply]
Sorry, another update:

I believe I have found the issue preventing me from pinging my test server. It's the firewall attached to my server (not the built in UFW or IPTables, the firewall attached to my Cloud server).

The thing is, the firewall has custom ports and their locked down to IPs so I can't see why that would prevent pings. I also have the standard 443, 22 and 80 open.

When I remove the server from firewall, the pings work. When I assign the firewall to the server, the pings stop working.

**Reply by SB8767** [reply]
Update: I removed the server from the firewall and set up iptables instead. I setup IPtables rules to replicate the rules that were configured in the firewall, then I ran the command `sudo iptables -A INPUT -j DROP`. This stopped the whole VPN from working.

I have two VPNs: one that works and one that isn't (and is the reason for this ticket).

The reason I set up iptables and removed the cloud firewall, is, because when I removed the server from the cloud firewall, the VPN started working and I could ping both side. When I added the Cloud firewall back in to the server, the ping stopped working.

I set up the firewall to allow TCP and UDP from any source and the PING still did not work. Once I remove the Cloud firewall from the server, the pings works, so that's why I looked at IPTables but the ping still does not work.
